tests/tsession.c

```diff
@@ -91,7 +101,12 @@ int main(int argc, char *argv[])
         for (int c = 0; c < CNUM; c++) {
             size_t size = EVP_PKEY_get_size(prikey);
             unsigned char sig[size];
-            ret = EVP_DigestSignFinal(sign_md[c], sig, &size);
+            if (mdname == NULL) {
+                ret =
+                    EVP_DigestSign(sign_md[c], sig, &size, data, sizeof(data));
```

Comment:
Given we do not really need multiple calls, would it mak sense to always just use the one shot call for all cases?

Reply:
Right, this would simplify things. I can change it as I hope no algorithm depends on having update.

Reply:
Changed.